### PR TITLE
Add a flag into gpcrondump to show the DDBoost configuration

### PR DESCRIPTION
**Pull request by tangp3**
information.

Authors: Christopher Hajas, Pengcheng Tang

### CODE REVIEW

File: gpMgmt/bin/gpcrondump [hunk 1]
```diff
@@ -801,6 +803,9 @@ class GpCronDump(Operation):
         #                        in this loop over the -x arguments
         final_exit_status = current_exit_status = 0
 
+        if self.ddboost_show_config:
+            self._show_ddboost_config()
```
Comment (larham) [inline gpMgmt/bin/gpcrondump]: would it be more clear to return immediately after this command?  should the _show* method return a non-zero error code if it fails?

File: gpMgmt/bin/gppylib/test/unit/test_cluster_gpcrondump.py [hunk 2]
```diff
@@ -1199,6 +1201,53 @@ def test_send_email00(self, mock1, MailDumpEvent):
...
+Default Backup Directory:MY_MFR
+Data Domain default log level:WARNING
+Data Domain Storage Unit:GPDB''', '', False, True)
+
+        options = GpCronDumpTestCase.Options()
+        options.ddboost_show_config = True
+
+        cron = GpCronDump(options, None)
+        logger.info = Mock()
+        cron._show_ddboost_config()
```
Comment (larham) [inline gpMgmt/bin/gppylib/test/unit/test_cluster_gpcrondump.py]: This unit tests calls a private method cron._show*

Is it possible to do the same test using a public method, such as supplying the options to the main() method?

File: gpMgmt/bin/gpcrondump (hunk 1 above)
Comment (tangp3) [inline gpMgmt/bin/gpcrondump]: displaying the config info does not necessarily have to prevent the normal dump, but returning before actual dump to separate them sounds good.

_show will raise exception if execution fails.

File: gpMgmt/bin/gppylib/test/unit/test_cluster_gpcrondump.py (hunk 2 above)
Comment (tangp3) [inline gpMgmt/bin/gppylib/test/unit/test_cluster_gpcrondump.py]: Making method as public is the best option for unit test.

**Comment by hlinnaka** [comment]
Seems straightforward. However, after PR #692, the lockbox file is human-readable anyway. Do we really need an option to print it out?

File: gpMgmt/bin/gppylib/test/unit/test_cluster_gpcrondump.py [hunk 3]
```diff
@@ -1199,6 +1201,53 @@ def test_send_email00(self, mock1, MailDumpEvent):
         cron = GpCronDump(options, None)
         cron._send_email(dump_database, current_exit_status, time_start, time_end)
 
+    @patch('gppylib.commands.base.Command.run')
```
Comment (kaknikhil) [inline gpMgmt/bin/gppylib/test/unit/test_cluster_gpcrondump.py]: maybe add a unit test for when the config is empty or not there at all

File: gpMgmt/bin/gpcrondump [hunk 4]
```diff
@@ -1217,6 +1222,22 @@ class GpCronDump(Operation):
 
         return pipe_list
 
+
+    def _show_ddboost_config(self):
+        cmdStr = 'gpddboost --show-config'
```
Comment (kaknikhil) [inline gpMgmt/bin/gpcrondump]: what happens if the user doesn't have a config ?

Comment (Chibin) [inline gpMgmt/bin/gpcrondump]: I believe line 1231 looks at the return code and raises an error, unless gpddboost somehow rerturns code 0 when the config doesn't exist... that would be awkward.

File: gpMgmt/bin/gppylib/test/unit/test_cluster_gpcrondump.py (hunk 3 above)
Comment (tangp3) [inline gpMgmt/bin/gppylib/test/unit/test_cluster_gpcrondump.py]: @hlinnaka 
This commit will be also put into 4.3.

@ivannovick
which version will be #692 targeting for?

Comment (hlinnaka) [inline gpMgmt/bin/gppylib/test/unit/test_cluster_gpcrondump.py]: #692 is definitely 5.0 material. We can't change the format of the lockbox file in a minor release.

Comment (tangp3) [inline gpMgmt/bin/gppylib/test/unit/test_cluster_gpcrondump.py]: We will add this PR both master and 4.3.

Comment (tangp3) [inline gpMgmt/bin/gppylib/test/unit/test_cluster_gpcrondump.py]: @kaknikhil 
In case the file is empty or not exist, gpddboost returns error, this can not be tested in unit, has to be tested in the ddboost functional tests part, we can add one there.

File: gpMgmt/bin/gpcrondump (hunk 4 above)
Comment (tangp3) [inline gpMgmt/bin/gpcrondump]: +1
